pkg/command/backup/logical.go

```diff
@@ -13,12 +13,17 @@ type logicalBackup struct {
 
 func (l *logicalBackup) BackupCommand(backup *mariadbv1alpha1.Backup,
 	mariadb *mariadbv1alpha1.MariaDB) *command.Command {
+	defOpts := "--single-transaction --events --routines --dump-slave=2 --master-data=2 --gtid --all-databases"
+	if l.BackupOpts.DumpOpts != "" {
+		defOpts = l.BackupOpts.DumpOpts
+	}
 	cmds := []string{
 		"set -euo pipefail",
 		"echo '💾 Taking logical backup'",
 		fmt.Sprintf(
-			"mysqldump %s --lock-tables --all-databases > %s",
+			"mariadb-backup %s %s > %s",
```

Comment:
Are `mysqldump` and `maria-backup` fully interchangeble? Asking this because after testing this locally it appears that `maria-backup` does not support all the option flags:
```bash
💾 Taking logical backup
mariadb-backup: unknown option '--single-transaction'

💾 Taking logical backup
mariadb-backup: unknown option '--events'

💾 Taking logical backup
mariadb-backup: unknown option '--routines'

💾 Taking logical backup
mariadb-backup: unknown option '--routines'

💾 Taking logical backup
mariadb-backup: unknown option '--all-databases'
```
See the backup container entrypoint for further detail:
```yaml
containers:
- args:
  - "set -euo pipefail;echo '\U0001F4BE Taking logical backup';mariadb-backup --user=${MARIADB_OPERATOR_USER}
    --password=${MARIADB_OPERATOR_PASSWORD} --host=mariadb.default.svc.cluster.local
    --port=3306 --single-transaction --events --routines --dump-slave=2 --master-data=2 --gtid --all-databases
    > /backup/backup.$(date -u +'%Y-%m-%dT%H:%M:%SZ').sql;echo '\U0001F9F9 Cleaning
    up old backups';find /backup -name *.sql -type f -mtime +30 -exec rm {} ';';echo
    '\U0001F4DC Backup history';find /backup -name *.sql -type f -printf '%f\n'
    | sort"
  command:
  - bash
  - -c
  env:
  - name: MARIADB_OPERATOR_USER
    value: root
  - name: MARIADB_OPERATOR_PASSWORD
    valueFrom:
      secretKeyRef:
        key: root-password
        name: mariadb
  image: mariadb:10.11.3
  imagePullPolicy: IfNotPresent
  name: mariadb
```
I'm affraid this will need some more refactors, for instance, the restore command won't work with `maria-backup` backups:
https://github.com/mariadb-operator/mariadb-operator/blob/8295963114e8305daeeab98e9472ccd8164b63aa/pkg/command/backup/logical.go#L49

Will note this down, and accommodate this in the roadmap along with the S3 storage support for backups, that will require a deep refactor of the backup/restore process, or even start from scratch:
https://github.com/mariadb-operator/mariadb-operator/blob/main/ROADMAP.md

Reply:
Ah I meant to do `mariadb-dump` here. Sorry, silly mistake. `mysql` -> `mariadb` too.

Reply:
Oh, right 👍🏻 do you mind pushing the change to your branch so we can trigger the tests again?